todo_app/data/user_login.py

```diff
@@ -0,0 +1,51 @@
+import requests
+import os
+import string, random
+
+def build_url(endpoint):
+    return str(os.getenv("GET_USER_IDENTITY_URL")) + endpoint
+
+
+def get_user_identity_endpoint():
+    
+    state_string  = ''.join(random.choices(string.ascii_letters + string.digits, k=10))
+
+    url = build_url('/authorize?' + 'client_id=' + str(os.getenv("CLIENT_ID")) + '&state=' + str(state_string))
+
+    return url
```

Comment:
You're building a state string here, but I can't see any evidence you actually use it for anything. 

If we're adding it for another layer of security, the intention there is to store the state in a way only we can (for example, in the user's cookies - e.g. using the `session` we used back in M1) and then check the state that GitHub redirects them with matches the one that we stored. That gives us confidence that it was us that kicked off the authentication process

Alternatively/additionally, it's sometimes used literally to store "state", like which page they were redirected from, although we can do so with cookies instead here

So while there's no harm with this use of state, it's worth noting that it doesn't seem to be achieving anything